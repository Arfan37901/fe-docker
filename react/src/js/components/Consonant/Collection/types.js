import {
    func,
    shape,
    number,
    arrayOf,
} from 'prop-types';

import { CardType } from '../types/card';

// eslint-disable-next-line import/prefer-default-export
export const CollectionType = {
    pages: number,
    resultsPerPage: number,
    cards: arrayOf(shape(CardType)),
    onCardBookmark: func.isRequired,
};
