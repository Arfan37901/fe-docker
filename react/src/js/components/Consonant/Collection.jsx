import React from 'react';
import PropTypes from 'prop-types';
import AspectRatio3to2Card from './Cards/3-2';
import AspectRatio1to1Card from './Cards/1-1';
import FullCard from './Cards/Full';

const DEFAULT_SHOW_ITEMS_PER_PAGE = 8;
const CARD_STYLE = {
    WIDE: '3:2',
    SQUARE: '1:1',
    FULL: 'full-card',
};

const Collection = (props) => {
    const {
        showItemsPerPage,
        pages,
        allowBookmarking,
        onCardBookmark,
        cardSavedIco,
        cardUnsavedIco,
        saveBookmarkText,
        unsaveBookmarkText,
        cardsStyle,
    } = props;
    let cards = [...props.cards];
    let cardsToShow = showItemsPerPage * pages;

    if (cardsToShow > cards.length) cardsToShow = cards.length;
    if (cardsToShow && cards.length > cardsToShow) cards = cards.slice(0, cardsToShow);

    return (
        <div className="consonant-card-collection">
            {cards.map((card) => {
                const type = cardsStyle && cardsStyle.toLowerCase() !== 'none' ? cardsStyle : card.cardStyle;

                if (type === CARD_STYLE.FULL) {
                    return (<FullCard
                        key={card.id}
                        {...card} />);
                } else if (type === CARD_STYLE.SQUARE) {
                    return (<AspectRatio1to1Card
                        key={card.id}
                        {...card} />);
                }

                return (<AspectRatio3to2Card
                    key={card.id}
                    {...card}
                    onClick={onCardBookmark}
                    cardSavedIco={cardSavedIco}
                    cardUnsavedIco={cardUnsavedIco}
                    allowBookmarking={allowBookmarking}
                    saveBookmarkText={saveBookmarkText}
                    unsaveBookmarkText={unsaveBookmarkText} />);
            })}
            <div className="consonant-card-collection--placeholder" />
            <div className="consonant-card-collection--placeholder" />
        </div>
    );
};

export default Collection;

Collection.propTypes = {
    showItemsPerPage: PropTypes.number,
    pages: PropTypes.number,
    cards: PropTypes.arrayOf(PropTypes.object),
    allowBookmarking: PropTypes.bool.isRequired,
    onCardBookmark: PropTypes.func.isRequired,
    cardSavedIco: PropTypes.string.isRequired,
    cardUnsavedIco: PropTypes.string.isRequired,
    saveBookmarkText: PropTypes.string,
    unsaveBookmarkText: PropTypes.string,
    cardsStyle: PropTypes.string,
};

Collection.defaultProps = {
    showItemsPerPage: DEFAULT_SHOW_ITEMS_PER_PAGE,
    pages: 1,
    cards: [],
    saveBookmarkText: 'Save card',
    unsaveBookmarkText: 'Unsave card',
    cardsStyle: 'none',
};
