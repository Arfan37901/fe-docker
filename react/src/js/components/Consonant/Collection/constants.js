import { DEFAULT_SHOW_ITEMS_PER_PAGE } from '../Helpers/constants';

// eslint-disable-next-line import/prefer-default-export
export const defaultProps = {
    pages: 1,
    cards: [],
    resultsPerPage: DEFAULT_SHOW_ITEMS_PER_PAGE,
};
