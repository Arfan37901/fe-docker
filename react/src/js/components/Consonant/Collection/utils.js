export const selector = ({
    pagination: { type },
    collection: { cardStyle },
}) => ({
    cardStyle,
    paginationType: type,
});

export const getFlatternProps = ({
    id,
    footer,
    ctaLink,
    isBookmarked,
    disableBookmarkIco,
    styles: {
        backgroundImage: cardImage,
    },
    contentArea: {
        title,
        description,
        detailText: label,
        dateDetailText: {
            endTime,
            startTime,
        },
    },
    overlays: {
        banner: {
            icon: bannerIcon,
            fontColor: bannerFontColor,
            description: bannerDescription,
            backgroundColor: bannerBackgroundColor,
        },
        videoButton: {
            url: videoURL,
        },
        logo: {
            src: logoSrc,
            alt: logoAlt,
            borderColor: logoBorderColor,
            backgroundColor: logoBackgroundColor,
        },
        label: {
            description: badgeText,
        },
    },
}) => ({
    id,
    title,
    label,
    footer,
    ctaLink,
    endTime,
    logoAlt,
    logoSrc,
    videoURL,
    badgeText,
    startTime,
    cardImage,
    bannerIcon,
    description,
    isBookmarked,
    logoBorderColor,
    bannerFontColor,
    bannerDescription,
    disableBookmarkIco,
    logoBackgroundColor,
    bannerBackgroundColor,
});
