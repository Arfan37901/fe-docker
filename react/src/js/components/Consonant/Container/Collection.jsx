import React from 'react';
import {
    func,
    shape,
    number,
    arrayOf,
} from 'prop-types';
import Collection from '@bit/fe-docker.consonant.collection';
import { cardType } from '../types/card';
import { useConfig } from '../Helpers/hooks';
import { DEFAULT_SHOW_ITEMS_PER_PAGE } from '../Helpers/constants';

import CustomCard from './CustomCard';

const collectionType = {
    pages: number,
    resultsPerPage: number,
    cards: arrayOf(shape(cardType)),
    onCardBookmark: func.isRequired,
};

const defaultProps = {
    pages: 1,
    cards: [],
    resultsPerPage: DEFAULT_SHOW_ITEMS_PER_PAGE,
};

/**
 * Contains a collection of cards (of different styles)
 *
 * @component
 * @example
 * const props= {
    resultPerPage: Int,
    pages: Int,
    onCardBookmark: Boolean,
    cards: [],
 * }
 * return (
 *   <Collection {...props}/>
 * )
 */
const CollectionContainer = ({
    cards,
    pages,
    resultsPerPage,
    onCardBookmark,
}) => {
    /**
     **** Authored Configs ****
     */
    const getConfig = useConfig();
    // const cardStyle = getConfig('collection', 'cardStyle');
    const dateFormat = getConfig('collection', 'i18n.prettyDateIntervalFormat');
    const locale = getConfig('language', '');
    const paginationType = getConfig('pagination', 'type');

    return (
        <Collection
            cards={cards}
            total={pages}
            locale={locale}
            cardStyle="custom"
            limit={resultsPerPage}
            dateFormat={dateFormat}
            CustomCard={CustomCard}
            paginationType={paginationType}
            onCardBookmark={onCardBookmark} />
    );
};

CollectionContainer.propTypes = collectionType;
CollectionContainer.defaultProps = defaultProps;

export default CollectionContainer;

