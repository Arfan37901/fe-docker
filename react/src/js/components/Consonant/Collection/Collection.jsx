/*eslint-disable */
import get from 'lodash/get';
import PropTypes from 'prop-types';
import React from 'react';
import { useConfig } from '../../../utils/hooks';
import AspectRatio1to1Card from '../Cards/1-1';
import AspectRatio3to2Card from '../Cards/3-2';
import FullCard from '../Cards/Full';

const DEFAULT_SHOW_ITEMS_PER_PAGE = 8;
const CARD_STYLE = {
    WIDE: '3:2',
    SQUARE: '1:1',
    FULL: 'full-card',
};

const Collection = (props) => {
    const {
        resultsPerPage,
        pages,
        onCardBookmark,
        cards,
    } = props;

    const getConfig = useConfig();
    const cardsStyle = getConfig('collection', 'cardStyle');
    const dateFormat = getConfig('collection', 'i18n.prettyDateIntervalFormat');
    const locale = getConfig('language', '');
    const paginationType = getConfig('pagination', 'type');

    let shownCards;

    if (paginationType === 'paginator') {
        shownCards = cards.slice(resultsPerPage * (pages - 1), resultsPerPage * pages);
    }
    else {
        shownCards = cards.slice(0, resultsPerPage * pages);
    }

    return shownCards.length > 0 && (
        <div data-testid="consonant-collection" className="consonant-card-collection">
            {shownCards.map((card) => {
                const type = cardsStyle && cardsStyle.toLowerCase() !== 'none' ? cardsStyle : get(card, 'styles.typeOverride');

                if (type === CARD_STYLE.FULL) {
                    return (<FullCard
                        key={card.id}
                        {...card} />);
                } else if (type === CARD_STYLE.SQUARE) {
                    return (<AspectRatio1to1Card
                        key={card.id}
                        {...card} />);
                }

                return (<AspectRatio3to2Card
                    key={card.id}
                    {...card}
                    onClick={onCardBookmark}
                    dateFormat={dateFormat}
                    locale={locale} />);
            })}
            <div className="consonant-card-collection--placeholder" />
            <div className="consonant-card-collection--placeholder" />
        </div>
    );
};

export default Collection;

Collection.propTypes = {
    resultsPerPage: PropTypes.number,
    pages: PropTypes.number,
    cards: PropTypes.arrayOf(PropTypes.object),
    onCardBookmark: PropTypes.func.isRequired,
};

Collection.defaultProps = {
    resultsPerPage: DEFAULT_SHOW_ITEMS_PER_PAGE,
    pages: 1,
    cards: [],
};
