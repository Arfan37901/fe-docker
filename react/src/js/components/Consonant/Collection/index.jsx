import React, { useMemo } from 'react';

import { If } from '../Common';
import { CollectionType } from './types';
import { defaultProps } from './constants';
import { CARD_STYLES } from '../Helpers/constants';
import { useConfigSelector } from '../Helpers/hooks';
import { getFlatternProps, selector } from './utils';
import {
    FullCard,
    WideCard,
    SquareCard,
} from '../Cards';

/**
 * Contains a collection of cards (of different styles)
 *
 * @component
 * @example
 * const props= {
    resultPerPage: Int,
    pages: Int,
    onCardBookmark: Boolean,
    cards: [],
 * }
 * return (
 *   <Collection {...props}/>
 * )
 */
const Collection = ({
    pages,
    cards,
    resultsPerPage,
    onCardBookmark,
}) => {
    /**
     **** Authored Configs ****
     */
    const {
        cardStyle,
        paginationType,
    } = useConfigSelector(selector);

    /**
     * The final cards to show in the collection
     * @type {Array}
     */
    const cardsToShow = useMemo(() => {
        /**
         * Whether the paginator component is being used
         * @type {Boolean}
         */
        const isPaginator = paginationType === 'paginator';
        const isLoadMore = paginationType === 'loadMore';

        /**
         * Total pages to show (used if paginator component is set)
         * @type {Number}
         */
        const totalPages = resultsPerPage * pages;

        /**
         * Current page (used if paginator component is authored)
         * @type {Number}
         */
        const currentPage = resultsPerPage * (pages - 1);

        if (isPaginator) {
            return cards.slice(currentPage, totalPages);
        }

        /**
         * Current page (used if load more button is authored)
         * @type {Number}
         */
        if (isLoadMore) {
            return cards.slice(0, resultsPerPage * pages);
        }

        return cards;
    }, [paginationType, resultsPerPage, pages, cards]);

    const renderCard = (card) => {
        const { id, styles: { typeOverride } } = card;

        const type = cardStyle || typeOverride;
        const props = {
            ...getFlatternProps(card),
            key: id,
        };

        switch (type) {
            case CARD_STYLES.FULL:
                return <FullCard {...props} />;
            case CARD_STYLES.SQUARE:
                return <SquareCard {...props} />;
            default:
                return <WideCard {...props} onClick={onCardBookmark} />;
        }
    };

    return (
        <If condition={Boolean(cardsToShow.length)}>
            <div
                data-testid="consonant-collection"
                className="consonant-card-collection">
                <div className="consonant-card-collection--inner">
                    {cardsToShow.map(renderCard)}
                    <div className="consonant-card-collection--placeholder" />
                    <div className="consonant-card-collection--placeholder" />
                </div>
            </div>
        </If>
    );
};

Collection.propTypes = CollectionType;
Collection.defaultProps = defaultProps;

export default Collection;

