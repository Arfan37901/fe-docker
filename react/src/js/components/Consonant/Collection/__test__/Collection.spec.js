import { screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Collection from '../Collection';

import { DEFAULT_PROPS } from '../../Testing/Constants/Collection';

import setup from '../../Testing/Utils/Settings';

const renderCollection = setup(Collection, DEFAULT_PROPS);

describe('Consonant/Collection', () => {
    test('should render 1 card if on page 1 and 1 card rendered per page', () => {
        renderCollection({ page: 1 });

        const cardElementList = screen.queryAllByTestId('consonant-OneHalfCard');

        expect(cardElementList).toHaveLength(1);
    });
    test('should be able to render all cards', () => {
        const {
            props: { cards },
        } = renderCollection({ resultsPerPage: Number.MAX_SAFE_INTEGER }, { pagination: { type: 'loadMore' } });

        const threeByTwoCards = screen.queryAllByTestId('consonant-OneHalfCard');
        const oneByOneCards = screen.queryAllByTestId('consonant-ThreeFourthCard');
        const fullCards = screen.queryAllByTestId('consonant-FullCard');

        const totalCards = threeByTwoCards.length + oneByOneCards.length + fullCards.length;
        expect(totalCards).toEqual(cards.length);
    });
});
