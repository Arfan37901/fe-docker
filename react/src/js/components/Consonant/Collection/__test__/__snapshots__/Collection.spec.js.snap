// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Collection Check snapshots should correct renders 1`] = `
<div
  className="consonant-card-collection"
  data-testid="consonant-collection"
>
  <div
    className="consonant-aspect-ratio-3-2-card"
    data-testid="consonant-card"
    id="6d6438c8-1"
  >
    <div
      className="consonant-aspect-ratio-3-2-card--img"
      data-testid="consonant-card--img"
      style={
        Object {
          "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
        }
      }
    >
      <span
        className="consonant-aspect-ratio-3-2-card--banner"
        data-testid="consonant-card--banner"
        style={
          Object {
            "backgroundColor": "#CB6F10",
            "color": "#FFFFFF",
          }
        }
      >
        <img
          alt=""
          className="consonant-aspect-ratio-3-2-card--banner-icon"
          data-testid="consonant-card--banner-icon"
          loading="lazy"
          src="some-banner-icon"
          width="8"
        />
        Some Banner Description
      </span>
      
      
    </div>
    <div
      className="consonant-aspect-ratio-3-2-card--inner"
    >
      
      <h2
        className="consonant-aspect-ratio-3-2-card--title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some Title 1",
          }
        }
      />
      <p
        className="consonant-aspect-ratio-3-2-card--text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars",
          }
        }
      />
      <div
        className="consonant-aspect-ratio-3-2-card--footer-wrapper"
      >
        <div
          className="consonant-aspect-ratio-3-2-card--footer-info"
        >
          <button
            className="consonant-aspect-ratio-3-2-card--footer-btn"
            data-testid="consonant-card__bookmark-button"
            data-tooltip-wrapper={true}
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="consonant-aspect-ratio-3-2-card--bookmark-ico"
              data-testid="bookmarks--ico"
              style={
                Object {
                  "backgroundImage": "url(https://upload.some-url)",
                }
              }
            />
            <span
              className="consonant-tooltip"
            >
              Save breakout session
            </span>
          </button>
          <span
            className="consonant-aspect-ratio-3-2-card--secondary-text"
          >
            Some secondary text
          </span>
        </div>
        <a
          className="consonant-aspect-ratio-3-2-card--btn"
          href="https://google.com"
          rel="noreferrer"
          tabIndex="0"
          target="_blank"
        >
          Some CTA
        </a>
      </div>
    </div>
  </div>
  <div
    className="consonant-card-collection--placeholder"
  />
  <div
    className="consonant-card-collection--placeholder"
  />
</div>
`;

exports[`Consonant/Collection Check snapshots should renders correctly with different card style 1`] = `
<div
  className="consonant-card-collection"
  data-testid="consonant-collection"
>
  <div
    className="consonant-aspect-ratio-3-2-card"
    data-testid="consonant-card"
    id="6d6438c8-1"
  >
    <div
      className="consonant-aspect-ratio-3-2-card--img"
      data-testid="consonant-card--img"
      style={
        Object {
          "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
        }
      }
    >
      <span
        className="consonant-aspect-ratio-3-2-card--banner"
        data-testid="consonant-card--banner"
        style={
          Object {
            "backgroundColor": "#CB6F10",
            "color": "#FFFFFF",
          }
        }
      >
        <img
          alt=""
          className="consonant-aspect-ratio-3-2-card--banner-icon"
          data-testid="consonant-card--banner-icon"
          loading="lazy"
          src="some-banner-icon"
          width="8"
        />
        Some Banner Description
      </span>
      
      
    </div>
    <div
      className="consonant-aspect-ratio-3-2-card--inner"
    >
      
      <h2
        className="consonant-aspect-ratio-3-2-card--title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some Title 1",
          }
        }
      />
      <p
        className="consonant-aspect-ratio-3-2-card--text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars",
          }
        }
      />
      <div
        className="consonant-aspect-ratio-3-2-card--footer-wrapper"
      >
        <div
          className="consonant-aspect-ratio-3-2-card--footer-info"
        >
          <button
            className="consonant-aspect-ratio-3-2-card--footer-btn"
            data-testid="consonant-card__bookmark-button"
            data-tooltip-wrapper={true}
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="consonant-aspect-ratio-3-2-card--bookmark-ico"
              data-testid="bookmarks--ico"
              style={
                Object {
                  "backgroundImage": "url(https://upload.some-url)",
                }
              }
            />
            <span
              className="consonant-tooltip"
            >
              Save breakout session
            </span>
          </button>
          <span
            className="consonant-aspect-ratio-3-2-card--secondary-text"
          >
            Some secondary text
          </span>
        </div>
        <a
          className="consonant-aspect-ratio-3-2-card--btn"
          href="https://google.com"
          rel="noreferrer"
          tabIndex="0"
          target="_blank"
        >
          Some CTA
        </a>
      </div>
    </div>
  </div>
  <div
    className="consonant-card-collection--placeholder"
  />
  <div
    className="consonant-card-collection--placeholder"
  />
</div>
`;

exports[`Consonant/Collection Check snapshots should renders correctly with different card style 2`] = `
<div
  className="consonant-card-collection"
  data-testid="consonant-collection"
>
  <div
    className="consonant-aspect-ratio-3-2-card"
    data-testid="consonant-card"
    id="6d6438c8-1"
  >
    <div
      className="consonant-aspect-ratio-3-2-card--img"
      data-testid="consonant-card--img"
      style={
        Object {
          "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
        }
      }
    >
      <span
        className="consonant-aspect-ratio-3-2-card--banner"
        data-testid="consonant-card--banner"
        style={
          Object {
            "backgroundColor": "#CB6F10",
            "color": "#FFFFFF",
          }
        }
      >
        <img
          alt=""
          className="consonant-aspect-ratio-3-2-card--banner-icon"
          data-testid="consonant-card--banner-icon"
          loading="lazy"
          src="some-banner-icon"
          width="8"
        />
        Some Banner Description
      </span>
      
      
    </div>
    <div
      className="consonant-aspect-ratio-3-2-card--inner"
    >
      
      <h2
        className="consonant-aspect-ratio-3-2-card--title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some Title 1",
          }
        }
      />
      <p
        className="consonant-aspect-ratio-3-2-card--text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars",
          }
        }
      />
      <div
        className="consonant-aspect-ratio-3-2-card--footer-wrapper"
      >
        <div
          className="consonant-aspect-ratio-3-2-card--footer-info"
        >
          <button
            className="consonant-aspect-ratio-3-2-card--footer-btn"
            data-testid="consonant-card__bookmark-button"
            data-tooltip-wrapper={true}
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="consonant-aspect-ratio-3-2-card--bookmark-ico"
              data-testid="bookmarks--ico"
              style={
                Object {
                  "backgroundImage": "url(https://upload.some-url)",
                }
              }
            />
            <span
              className="consonant-tooltip"
            >
              Save breakout session
            </span>
          </button>
          <span
            className="consonant-aspect-ratio-3-2-card--secondary-text"
          >
            Some secondary text
          </span>
        </div>
        <a
          className="consonant-aspect-ratio-3-2-card--btn"
          href="https://google.com"
          rel="noreferrer"
          tabIndex="0"
          target="_blank"
        >
          Some CTA
        </a>
      </div>
    </div>
  </div>
  <div
    className="consonant-card-collection--placeholder"
  />
  <div
    className="consonant-card-collection--placeholder"
  />
</div>
`;

exports[`Consonant/Collection Check snapshots should renders correctly with different card style 3`] = `
<div
  className="consonant-card-collection"
  data-testid="consonant-collection"
>
  <div
    className="consonant-aspect-ratio-3-2-card"
    data-testid="consonant-card"
    id="6d6438c8-1"
  >
    <div
      className="consonant-aspect-ratio-3-2-card--img"
      data-testid="consonant-card--img"
      style={
        Object {
          "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
        }
      }
    >
      <span
        className="consonant-aspect-ratio-3-2-card--banner"
        data-testid="consonant-card--banner"
        style={
          Object {
            "backgroundColor": "#CB6F10",
            "color": "#FFFFFF",
          }
        }
      >
        <img
          alt=""
          className="consonant-aspect-ratio-3-2-card--banner-icon"
          data-testid="consonant-card--banner-icon"
          loading="lazy"
          src="some-banner-icon"
          width="8"
        />
        Some Banner Description
      </span>
      
      
    </div>
    <div
      className="consonant-aspect-ratio-3-2-card--inner"
    >
      
      <h2
        className="consonant-aspect-ratio-3-2-card--title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some Title 1",
          }
        }
      />
      <p
        className="consonant-aspect-ratio-3-2-card--text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars",
          }
        }
      />
      <div
        className="consonant-aspect-ratio-3-2-card--footer-wrapper"
      >
        <div
          className="consonant-aspect-ratio-3-2-card--footer-info"
        >
          <button
            className="consonant-aspect-ratio-3-2-card--footer-btn"
            data-testid="consonant-card__bookmark-button"
            data-tooltip-wrapper={true}
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="consonant-aspect-ratio-3-2-card--bookmark-ico"
              data-testid="bookmarks--ico"
              style={
                Object {
                  "backgroundImage": "url(https://upload.some-url)",
                }
              }
            />
            <span
              className="consonant-tooltip"
            >
              Save breakout session
            </span>
          </button>
          <span
            className="consonant-aspect-ratio-3-2-card--secondary-text"
          >
            Some secondary text
          </span>
        </div>
        <a
          className="consonant-aspect-ratio-3-2-card--btn"
          href="https://google.com"
          rel="noreferrer"
          tabIndex="0"
          target="_blank"
        >
          Some CTA
        </a>
      </div>
    </div>
  </div>
  <div
    className="consonant-card-collection--placeholder"
  />
  <div
    className="consonant-card-collection--placeholder"
  />
</div>
`;

exports[`Consonant/Collection Check snapshots should renders correctly with different card style 4`] = `
<div
  className="consonant-card-collection"
  data-testid="consonant-collection"
>
  <div
    className="consonant-aspect-ratio-3-2-card"
    data-testid="consonant-card"
    id="6d6438c8-1"
  >
    <div
      className="consonant-aspect-ratio-3-2-card--img"
      data-testid="consonant-card--img"
      style={
        Object {
          "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
        }
      }
    >
      <span
        className="consonant-aspect-ratio-3-2-card--banner"
        data-testid="consonant-card--banner"
        style={
          Object {
            "backgroundColor": "#CB6F10",
            "color": "#FFFFFF",
          }
        }
      >
        <img
          alt=""
          className="consonant-aspect-ratio-3-2-card--banner-icon"
          data-testid="consonant-card--banner-icon"
          loading="lazy"
          src="some-banner-icon"
          width="8"
        />
        Some Banner Description
      </span>
      
      
    </div>
    <div
      className="consonant-aspect-ratio-3-2-card--inner"
    >
      
      <h2
        className="consonant-aspect-ratio-3-2-card--title"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Some Title 1",
          }
        }
      />
      <p
        className="consonant-aspect-ratio-3-2-card--text"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars",
          }
        }
      />
      <div
        className="consonant-aspect-ratio-3-2-card--footer-wrapper"
      >
        <div
          className="consonant-aspect-ratio-3-2-card--footer-info"
        >
          <button
            className="consonant-aspect-ratio-3-2-card--footer-btn"
            data-testid="consonant-card__bookmark-button"
            data-tooltip-wrapper={true}
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="consonant-aspect-ratio-3-2-card--bookmark-ico"
              data-testid="bookmarks--ico"
              style={
                Object {
                  "backgroundImage": "url(https://upload.some-url)",
                }
              }
            />
            <span
              className="consonant-tooltip"
            >
              Save breakout session
            </span>
          </button>
          <span
            className="consonant-aspect-ratio-3-2-card--secondary-text"
          >
            Some secondary text
          </span>
        </div>
        <a
          className="consonant-aspect-ratio-3-2-card--btn"
          href="https://google.com"
          rel="noreferrer"
          tabIndex="0"
          target="_blank"
        >
          Some CTA
        </a>
      </div>
    </div>
  </div>
  <div
    className="consonant-card-collection--placeholder"
  />
  <div
    className="consonant-card-collection--placeholder"
  />
</div>
`;
